### DISCUSSION TITLE
[node] pipeline callback should not be optional

### DISCUSSION THREAD

**Discussion by jhmen** [DefinitelyTyped/DefinitelyTyped#67345] [Issues with a @types package]
In the types for the node pipeline function (stream) the callback is marked as optional, but Node requires a callback or it will throw an error.

I have created a reproduce repo here: https://github.com/jhmen/pipeline-wrong-typing. If you ran this, you will get the error: ERR_INVALID_ARG_TYPE. But no typescript errors.

The solution would be to mark the callback as not optional

[a comment by an automated account is not shown]

**Comment by ghost** [comment]
Fixed in #69498
